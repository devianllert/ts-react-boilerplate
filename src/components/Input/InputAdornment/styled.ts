/* eslint-disable import/prefer-default-export */
import styled from 'styled-components';

import { spacings } from '../../../design/spacings';

interface AdornmentProps {
  position: 'start' | 'end';
  variant?: 'standard' | 'outlined' | 'filled';
  disablePointerEvents: boolean;
}

export const Adornment = styled.div<AdornmentProps>`
  display: flex;
  align-items: center;

  max-height: 2em;

  white-space: nowrap;

  margin-right: ${({ position }): string => (position === 'start' ? spacings[1] : '0')} ;
  margin-left: ${({ position }): string => (position === 'end' ? spacings[1] : '0')} ;

  pointer-events: ${({ disablePointerEvents }): string => (disablePointerEvents ? 'none' : 'auto')}
`;
