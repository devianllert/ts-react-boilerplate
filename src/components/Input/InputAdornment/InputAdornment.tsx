import React, {
  ReactElement,
  ReactNode,
  HTMLAttributes,
} from 'react';

import Typography from '../../Typography';

import * as S from './styled';

interface InputAdornmentProps extends HTMLAttributes<HTMLDivElement> {
  /**
   * The content of the component, normally an `IconButton` or string.
   */
  children: ReactNode;
  /**
   * Disable pointer events on the root.
   * This allows for the content of the adornment to focus the input on click.
   */
  disablePointerEvents?: boolean;
  /**
   * If children is a string then disable wrapping in a Typography component.
   */
  disableTypography?: boolean;
  /**
   * The position this adornment should appear relative to the `Input`.
   */
  position: 'start' | 'end';
  /**
   * The variant to use.
   */
  variant?: 'standard' | 'outlined' | 'filled';
}

const InputAdornment = (props: InputAdornmentProps): ReactElement => {
  const {
    children,
    className,
    disablePointerEvents = false,
    disableTypography = false,
    position,
    variant,
    ...other
  } = props;

  return (
    <S.Adornment
      variant={variant}
      position={position}
      disablePointerEvents={disablePointerEvents}
      className={className}
      // eslint-disable-next-line react/jsx-props-no-spreading
      {...other}
    >
      {typeof children === 'string' && !disableTypography ? (
        <Typography color="textSecondary">{children}</Typography>
      ) : (
        children
      )}
    </S.Adornment>
  );
};

export default InputAdornment;
