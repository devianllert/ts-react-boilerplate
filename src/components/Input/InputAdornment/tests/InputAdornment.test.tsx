import React from 'react';
import { render, screen } from '@testing-library/react';

import InputAdornment from '../InputAdornment';

describe('<InputAdornment />', () => {
  it('should wrap text children in a Typography', () => {
    render(<InputAdornment position="start">foo</InputAdornment>);

    const typography = screen.getByText('foo');

    expect(typography).toHaveStyleRule('font-size', '1.6rem');
  });

  it('should have the disabled pointer events class when disabledPointerEvents true', () => {
    render(
      <InputAdornment disablePointerEvents position="start">
        foo
      </InputAdornment>,
    );
    const adornment = screen.getByText('foo').parentElement;

    expect(adornment).toHaveStyleRule('pointer-events', 'none');
  });
});
