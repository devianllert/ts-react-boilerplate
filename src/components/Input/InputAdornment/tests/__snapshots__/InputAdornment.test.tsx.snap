// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputAdornment /> should have the root and end class when position is end 1`] = `
.c1 {
  margin: 0;
  text-align: inherit;
  font-size: 1.6rem;
  font-weight: 400;
  -webkit-letter-spacing: 0.15px;
  -moz-letter-spacing: 0.15px;
  -ms-letter-spacing: 0.15px;
  letter-spacing: 0.15px;
  line-height: 1.5;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  max-height: 2em;
  white-space: nowrap;
  margin-right: 0;
  margin-left: 8px;
  pointer-events: auto;
}

<div
  class="c0"
  data-testid="adornment"
>
  <p
    class="c1"
    color="textSecondary"
    display="initial"
  >
    foo
  </p>
</div>
`;

exports[`<InputAdornment /> should have the root and start class when position is start 1`] = `
.c1 {
  margin: 0;
  text-align: inherit;
  font-size: 1.6rem;
  font-weight: 400;
  -webkit-letter-spacing: 0.15px;
  -moz-letter-spacing: 0.15px;
  -ms-letter-spacing: 0.15px;
  letter-spacing: 0.15px;
  line-height: 1.5;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  max-height: 2em;
  white-space: nowrap;
  margin-right: 8px;
  margin-left: 0;
  pointer-events: auto;
}

<div
  class="c0"
  data-testid="adornment"
>
  <p
    class="c1"
    color="textSecondary"
    display="initial"
  >
    foo
  </p>
</div>
`;
